test/Npgsql.Tests/Types/TypeHandlerTestBase.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Threading.Tasks;
+using NpgsqlTypes;
+using NUnit.Framework;
+using static Npgsql.Tests.TestUtil;
+
+namespace Npgsql.Tests.Types
+{
+    public abstract class TypeHandlerTestBase<T> : MultiplexingTestBase
+    {
+        readonly string? _typeName;
+        readonly string? _minVersion;
+
+        protected TypeHandlerTestBase(MultiplexingMode multiplexingMode, string? typeName = null, string? minVersion = null)
+            : base(multiplexingMode) => (_typeName, _minVersion) = (typeName, minVersion);
+
+        [Test]
+        [TestCaseSource("TestCases")]
+        public async Task Read(string query, T expected)
+        {
+            using var conn = await OpenConnectionAsync();
+            using var cmd = new NpgsqlCommand($"SELECT {query}", conn);
+
+            if (_minVersion is string minVersion)
```

Comment:
This check can move to a [OneTimeSetUp] method (in Write too)

Reply:
Not possible due to parametrization.

Reply:
There was a different issue, NUnit isn't able to construct an instance if its constructor has more than one argument.